### PR TITLE
Add example of execution limits using fuel consumption

### PR DESCRIPTION
**Pull request by ryanpbrewster**
This is a quick example of using the "fuel consumption" logic introduced in https://github.com/bytecodealliance/wasmtime/pull/2611/.

It computes larger and larger Fibonacci numbers (using the expensive recursive algorithm) until the computation requires more than 10k fuel.

### CODE REVIEW

File: examples/fuel.wat [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      end
+      i32.const 1
+    else
+      local.get 0
+    end
+    local.get 1
+    i32.add
+  )
+  (export "fibonacci" (func $fibonacci))
+)
```
Comment (alexcrichton) [inline examples/fuel.wat:36]: Could this be written in either the "naive" form (which should be easier to read) or as a Rust program where the source is more readable? I'm a bit worried that digesting this `*.wat` file is a little complicated.

**Review by alexcrichton — COMMENTED** [review]
Thanks for this!

File: examples/fuel.wat (hunk 1 above)
Comment (ryanpbrewster) [reply examples/fuel.wat:13]: I took a stab at re-implementing this using a more natural style. PTAL.

**Comment by alexcrichton** [comment]
Ah it looks like CI is failing when it's running the examples, as it expects that all examples pass successfully. Could the example be updated to not return a failure code? (e.g. catch the error)

Also I think our CI verifies that all examples are available in both C and Rust, so I think that at least a stub program will be needed for a C example?

**Comment by ryanpbrewster** [comment]
Ah, that could be it. Thanks for the pointer, adding a C stub now.

**Comment by ryanpbrewster** [comment]
Added a C example. I'm not particularly confident in the error handling and cleanup code there. I think I handle the error codes from `wasm_store_add_fuel` (and testing that if the config does not enable fuel consumption that those calls fail and exit).

I'm not sure about the semantics of the `bool` that `wasm_store_fuel_consumed` returns, so I just ignored it. Should I be asserting that it's true?

**Comment by alexcrichton** [comment]
That all looks great to me, thanks!

**Pull request merged**